### DISCUSSION TITLE
ffmpeg的路径的获取方式（下策

### DISCUSSION THREAD

**Discussion by nanarino** [yutto-dev/yutto#79] [Ideas]
可以多装一个imageio，这样找不到ffmpeg的时候通过下载pypi上的高版本的imageio包间接下载ffmpeg。

然后通过imageio_ffmpeg.get_ffmpeg_exe()得到路径：

```
    >>> import imageio_ffmpeg
    >>> imageio_ffmpeg.get_ffmpeg_exe()
    'C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\imageio_ffmpeg\\binaries\\ffmpeg-win64-v4.2.2.exe'
```

低版本的imageio曾经有一种显示下载的

```
    import imageio
    imageio.plugins.ffmpeg.download()
```
🥰这样或许对不懂环境变量的小白更友好

**Comment by SigureMo** [comment]
目前还没遇到在 FFmpeg 配置上出问题的，因此暂不考虑

自行配置一方面是考虑到这样能确保获取较新版本（现在最新版本是 5.x，而你这里展示的还是 4.x，而 yutto 可能在一定程度上需要 FFmpeg5.x 的特性），另一方面考虑到它下载速度的问题，毕竟在我还是小白的时候就有过因为使用这种包直接下载结果半天都下载不下来的经历（好像是 FFmpeg 相关的包）

另外，很多人其实都有安装和使用 FFmpeg，为什么要额外用库再去安装一个呢？至少我是不愿意的，而且一件完全可以手动解决的事情却去调用一个无关紧要的库不是明智之举

另外不懂环境变量就别用了（随便查查就能知道的东西，我也有写详细的步骤）……我回答问题也头疼……
